### ISSUE TITLE
"WARNING [env_manager.py:101] Agent manager was not created for behavior id"

### ISSUE THREAD

**Issue by starman12** [Unity-Technologies/ml-agents#3806]
Hey! I'm trying to train a self-play ai with team ids 0 and 1, but I keep getting this error in anaconda everytime, I tried to look in the env_manager.py file but couldn't seem to understand it. Would really appreciate it if someone can point me to a documentation or a solution 😀

**Comment by anupambhatnagar** [comment]
@starman12 your comment does not state the error message. moreover we reserve github for feature requests and bugs. could you please post your error message in the ML-Agents forum. https://forum.unity.com/forums/ml-agents.453/

**Comment by chairiq** [comment]
Hi I get a similar warning when I run my game, which is a board game of two agents. The RedAgent (TeamID=1) and the BlueAgent (TeamID=2)



(MLAgentUnityEnv) user-mbp:myData user$ mlagents-learn notebooks/trainer_config.yaml --run-id=RLGame
WARNING:tensorflow:From /Users/user/Programming/MLAgentUnityEnv/lib/python3.7/site-packages/tensorflow_core/python/compat/v2_compat.py:65: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term


                        ▄▄▄▓▓▓▓
                   ╓▓▓▓▓▓▓█▓▓▓▓▓
              ,▄▄▄m▀▀▀'  ,▓▓▓▀▓▓▄                           ▓▓▓  ▓▓▌
            ▄▓▓▓▀'      ▄▓▓▀  ▓▓▓      ▄▄     ▄▄ ,▄▄ ▄▄▄▄   ,▄▄ ▄▓▓▌▄ ▄▄▄    ,▄▄
          ▄▓▓▓▀        ▄▓▓▀   ▐▓▓▌     ▓▓▌   ▐▓▓ ▐▓▓▓▀▀▀▓▓▌ ▓▓▓ ▀▓▓▌▀ ^▓▓▌  ╒▓▓▌
        ▄▓▓▓▓▓▄▄▄▄▄▄▄▄▓▓▓      ▓▀      ▓▓▌   ▐▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▌   ▐▓▓▄ ▓▓▌
        ▀▓▓▓▓▀▀▀▀▀▀▀▀▀▀▓▓▄     ▓▓      ▓▓▌   ▐▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▌    ▐▓▓▐▓▓
          ^█▓▓▓        ▀▓▓▄   ▐▓▓▌     ▓▓▓▓▄▓▓▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▓▄    ▓▓▓▓`
            '▀▓▓▓▄      ^▓▓▓  ▓▓▓       └▀▀▀▀ ▀▀ ^▀▀    `▀▀ `▀▀   '▀▀    ▐▓▓▌
               ▀▀▀▀▓▄▄▄   ▓▓▓▓▓▓,                                      ▓▓▓▓▀
                   `▀█▓▓▓▓▓▓▓▓▓▌
                        ¬`▀▀▀█▓

        
 Version information:
  ml-agents: 0.15.1,
  ml-agents-envs: 0.15.1,
  Communicator API: 0.15.0,
  TensorFlow: 2.0.1
2020-04-30 18:45:21 INFO [environment.py:132] Listening on port 5004. Start training by pressing the Play button in the Unity Editor.
2020-04-30 18:45:27 INFO [environment.py:160] Connected to Unity environment with package version 0.15.1-preview and communication version 0.15.0
2020-04-30 18:45:27 INFO [environment.py:305] Connected new brain:
RedAgent?team=1
2020-04-30 18:45:27 INFO [trainer_controller.py:167] Hyperparameters for the GhostTrainer of brain RedAgent: 
	trainer:	ppo
	batch_size:	2048
	beta:	0.001
	buffer_size:	20480
	epsilon:	0.2
	hidden_units:	5
	lambd:	0.99
	learning_rate:	0.0003
	learning_rate_schedule:	constant
	max_steps:	6.0e4
	memory_size:	128
	normalize:	True
	num_epoch:	2
	num_layers:	2
	time_horizon:	1000
	sequence_length:	64
	summary_freq:	1000
	use_recurrent:	False
	vis_encode_type:	simple
	reward_signals:	
	  extrinsic:	
	    strength:	1.0
	    gamma:	0.99
	summary_path:	RLGame_RedAgent
	model_path:	./models/RLGame/RedAgent
	keep_checkpoints:	5
	self_play:	
	  window:	10
	  play_against_latest_model_ratio:	0.5
	  save_steps:	1000
	  swap_steps:	25000
	  team_change:	200000
2020-04-30 18:45:27.497444: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-04-30 18:45:27.510553: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7fcbef0cfa00 executing computations on platform Host. Devices:
2020-04-30 18:45:27.510571: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Host, Default Version
2020-04-30 18:45:30 INFO [environment.py:305] Connected new brain:
BlueAgent?team=2
2020-04-30 18:45:30 WARNING [env_manager.py:101] Agent manager was not created for behavior id BlueAgent?team=2.
2020-04-30 18:45:30 INFO [trainer_controller.py:167] Hyperparameters for the GhostTrainer of brain BlueAgent: 
	trainer:	ppo
	batch_size:	2048
	beta:	0.001
	buffer_size:	20480
	epsilon:	0.2
	hidden_units:	5
	lambd:	0.99
	learning_rate:	0.0003
	learning_rate_schedule:	constant
	max_steps:	6.0e4
	memory_size:	128
	normalize:	True
	num_epoch:	2
	num_layers:	2
	time_horizon:	1000
	sequence_length:	64
	summary_freq:	1000
	use_recurrent:	False
	vis_encode_type:	simple
	reward_signals:	
	  extrinsic:	
	    strength:	1.0
	    gamma:	0.99
	summary_path:	RLGame_BlueAgent
	model_path:	./models/RLGame/BlueAgent
	keep_checkpoints:	5
	self_play:	
	  play_against_latest_model_ratio:	0.5
	  save_steps:	1000
	  swap_steps:	25000
	  team_change:	200000
2020-04-30 18:45:45 INFO [trainer.py:214] RLGame: BlueAgent: Step: 1000. Time Elapsed: 14.710 s Mean Reward: 0.387. Std of Reward: 0.539. Not Training.
2020-04-30 18:45:45 INFO [trainer.py:103]  Learning brain BlueAgent?team=2 ELO: 1201.496
Mean Opponent ELO: 1199.864 Std Opponent ELO: 0.430
2020-04-30 18:45:47 INFO [trainer.py:214] RLGame: RedAgent: Step: 1000. Time Elapsed: 20.199 s Mean Reward: -0.145. Std of Reward: 0.000. Not Training.
2020-04-30 18:45:47 INFO [trainer.py:103]  Learning brain RedAgent?team=1 ELO: 1200.500
Mean Opponent ELO: 1199.955 Std Opponent ELO: 0.144


In addition to that when I export my game and try it with the python code the python code recognizes only the one agent

`
import matplotlib.pyplot as plt
import numpy as np
import sys

from mlagents_envs.environment import UnityEnvironment
from mlagents_envs.side_channel.engine_configuration_channel import EngineConfig, EngineConfigurationChannel

print("Python version:")
print(sys.version)

if (sys.version_info[0] < 3):
raise Exception("ERROR: ML-Agents Toolkit (v0.3 onwards) requires Python 3")

env_name = "RLGame"
engine_configuration_channel = EngineConfigurationChannel()
env = UnityEnvironment(file_name=env_name, side_channels = [engine_configuration_channel])

#Reset the environment
env.reset()

group_name = env.get_agent_groups()
print(group_name)
`

**Comment by hecticmonke** [comment]
Did anybody solve the problem?
